File: app/src/main/java/com/ignis/android_cleanarchitecture/domain/usecase/AboutUseCaseImpl.java
package com.ignis.android_cleanarchitecture.domain.usecase;

import com.ignis.android_cleanarchitecture.domain.repository.AboutRepository;

import java.util.Calendar;

import rx.Observable;

/**
 * About UseCase Implement
 *
 * @author Kensuke
 */
public class AboutUseCaseImpl implements AboutUseCase {

    private AboutRepository aboutRepository;

    public AboutUseCaseImpl(AboutRepository aboutRepository) {
        this.aboutRepository = aboutRepository;
    }

    @Override
    public Observable<String> getPlayStoreUrl() {
        return aboutRepository.getPlayStoreUrl();
    }

    @Override
    public Observable<String> getMailUrl() {
        return aboutRepository.getMailUrl();
    }

    @Override
    public Observable<String> getWebSiteUrl() {
        return aboutRepository.getWebSiteUrl();
    }

    @Override
    public Observable<String> getCurrentVersion() {
        return aboutRepository.getCurrentVersion();
    }

    @Override
    public Observable<Integer> getCurrentYear() {
        return Observable.create((Observable.OnSubscribe<Integer>) subscriber -> {
            Calendar calendar = Calendar.getInstance();
            subscriber.onNext(calendar.get(Calendar.YEAR));
            subscriber.onCompleted();
        });
    }

}

File: app/src/main/java/com/kazakago/cleanarchitecture/domain/usecase/AboutUseCaseImpl.kt
package com.kazakago.cleanarchitecture.domain.usecase

import com.kazakago.cleanarchitecture.domain.repository.AboutRepository
import java.util.*

/**
 * About UseCase Implement
 *
 * @author Kensuke
 */
class AboutUseCaseImpl(private val aboutRepository: AboutRepository) : AboutUseCase {

    override val playStoreUrl: String
        get() = aboutRepository.playStoreUrl

    override val mailUrl: String
        get() = aboutRepository.mailUrl

    override val webSiteUrl: String
        get() = aboutRepository.webSiteUrl

    override val currentVersion: String
        get() = aboutRepository.currentVersion

    override val currentYear: Int
        get() {
            val calendar = Calendar.getInstance()
            return calendar.get(Calendar.YEAR)
        }

}
